### PR TITLE
Use julia 1.10 as default in CI

### PR DESCRIPTION
**Pull request by lgoettgens**
Merging this needs also an update to the list of required CI jobs.

### CODE REVIEW

File: .github/workflows/CILong.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
       fail-fast: false
       matrix:
         julia-version:
-          - 1.9
+          - 1.10
```
Comment (lgoettgens) [inline .github/workflows/CILong.yml:25]:
```suggestion
          - '1.10'
```

File: .github/workflows/CILong.yml [hunk 2]
```diff
@@ -62,7 +62,7 @@ jobs:
...
```
Comment (lgoettgens) [inline .github/workflows/CILong.yml:65]:
```suggestion
          - '1.10'
```

**Comment by lgoettgens** [comment]
This now looks good. Somehow, the 1.10 was parsed as 1.1 before.

**Comment by thofma** [comment]
Thanks! I will merge once green. Afterwards I can adjust the "required" thingies.

**Pull request merged**